sklearn/tests/test_calibration.py

```diff
@@ -179,6 +179,28 @@ def test_sample_weight(data, method, ensemble):
     assert diff > 0.1
 
 
+@pytest.mark.parametrize("method", ["sigmoid", "isotonic"])
+@pytest.mark.parametrize("ensemble", [True, False])
+def test_sample_weight_class_imbalanced(data, method, ensemble):
+    X, y = make_blobs((100, 1000), center_box=(-1, 1), random_state=42)
+
+    # Compute weigths to compensate the unbalance of the dataset
+    sample_weight = 9 * (y == 0) + 1
+
+    X_train, X_test, y_train, y_test, sw_train, sw_test = train_test_split(
+        X, y, sample_weight, random_state=42
+    )
+
+    base_estimator = LinearSVC(random_state=42)
+    calibrated_clf = CalibratedClassifierCV(
+        base_estimator, method=method, ensemble=ensemble
+    )
+    calibrated_clf.fit(X_train, y_train, sample_weight=sw_train)
+    predictions = calibrated_clf.predict_proba(X_test)[:, 1]
+
+    assert brier_score_loss(y_test, predictions, sample_weight=sw_test) < 0.2
```

Comment:
I am thinking that we could compare the performance of the `ensemble=True` and `ensemble=False` and check that they lead to a close score. We would remove the parametrization on the ensemble then.
We could still keep this check here.

Reply:
Makes sense. I'll do it.

Reply:
This is also nice to check the dependence here. You can add a new test to test this dependence instead of changing this one.

Reply:
done. (see test_sample_weight_class_imbalanced_ensemble_equivalent)